### PR TITLE
fix reisze whisper from low retention to high retention will lose some data

### PR DESCRIPTION
**Pull request by Xu-Wentao**
when resize whisper file from [60s:30d,300s:365d]  to [60s:365d] will lose some 60s data

### CODE REVIEW

File: bin/whisper-resize.py [hunk 1]
```diff
@@ -107,20 +107,22 @@
   # This is where data will be interpolated (best effort)
   print('Migrating data with aggregation...')
   all_datapoints = []
-  for archive in old_archives:
+  for archive in sorted(old_archives, key=lambda x: x['secondsPerPoint']):
```
Comment (deniszh) [inline bin/whisper-resize.py:110]: Looks logical, indeed

File: bin/whisper-resize.py [hunk 2]
```diff
@@ -107,20 +107,22 @@
...
     # Loading all datapoints into memory for fast querying
     timeinfo, values = archive['data']
     new_datapoints = list(zip(range(*timeinfo), values))
+    new_datapoints.reverse()
```
Comment (deniszh) [inline bin/whisper-resize.py:114]: A-ha, I see where it's going, nice.

File: bin/whisper-resize.py [hunk 3]
```diff
@@ -107,20 +107,22 @@
...
       for i, (timestamp, value) in enumerate(new_datapoints):
-        if timestamp > last_timestamp:
+        if timestamp < last_timestamp:
           slice_end = i
           break
-      all_datapoints += new_datapoints[i:]
+      all_datapoints += new_datapoints[slice_end:]
     else:
       all_datapoints += new_datapoints
+    all_datapoints.reverse()
```
Comment (deniszh) [inline bin/whisper-resize.py:125]: OK, looks plausible.

**Comment by Xu-Wentao** [comment]
I have push a new test case to GitHub and fix a Indentation bug, please review the pr
When I write the test case I use os.system to call whisper-resize function, I think it’s not good can you give me some suggestion about how to call whisper-resize function currently?
在 2020年5月8日 +0800 PM2:59，Denis Zhdanov <notifications@github.com>，写道：

**Comment by piotr1212** [comment]
The cli tools are a bit hard to test. The dash in the filenames makes it impossible to load the files and then most of the tools don't have code in functions. 

We can use console_scripts to fix the filenames now we switched to setuptools. But splitting all code  in functions is just a lot of boring work.

Anyway, I'm ok with current os.system method

**Comment by deniszh** [comment]
Thanks a lot, @Xu-Wentao !
Merging

**Pull request merged**